src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -10,64 +10,137 @@ namespace System.Net
     internal class HeaderInfoTable
     {
         private static Hashtable HeaderHashTable;
-        private static HeaderInfo UnknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false);
+        private static HeaderInfo UnknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false, SingleParser);
+        private static HeaderParser SingleParser = new HeaderParser(ParseSingleValue);
+        private static HeaderParser MultiParser = new HeaderParser(ParseMultiValue);
+        private static HeaderParser SetCookieParser = new HeaderParser(ParseSetCookieValue);
```

Comment:
These should all be `readonly` and the coding style says to prefix with `s_` (e.g. `s_singleParser`).

Reply:
These can also be simplified, e.g.:

```c#
private static readonly Func<string, string[]> s_singleParser = value => new[] { value };
private static readonly Func<string, string[]> s_multiParser = value => ParseValueHelper(value, isSetCookie: false);
private static readonly Func<string, string[]> s_setCookieParser = value => ParseValueHelper(value, isSetCookie: true);
private static readonly HeaderInfo s_unknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false, s_singleParser);
private static readonly Hashtable s_headerHashTable;
```